### PR TITLE
Android Nougat bugfix

### PR DESCRIPTION
**Pull request by nicolabeghin**
Fixes issue #74 
Fixes AutoUpdateAPK on Android Nougat (7), as `Context.MODE_WORLD_READABLE` throws `SecurityException`

### CODE REVIEW

File: src/com/lazydroid/autoupdateapk/AutoUpdateApk.java [hunk 1]
```diff
@@ -338,7 +340,12 @@ private static String getResponseText(InputStream inStream) {
 
 						in = new BufferedInputStream(conn.getInputStream());
 						String fname = result[1].substring(result[1].lastIndexOf('/')+1);
-						FileOutputStream out = context.openFileOutput( fname, Context.MODE_WORLD_READABLE);
+						FileOutputStream out = null;
```
Comment (duchuyctlk) [inline src/com/lazydroid/autoupdateapk/AutoUpdateApk.java]: For a shorter statement: 😄 

```java
FileOutputStream out = context.openFileOutput(fname, 
        Build.VERSION.SDK_INT >= Build.VERSION_CODES.N ? 
        Context.MODE_PRIVATE : Context.MODE_WORLD_READABLE);
```

Comment (nicolabeghin) [inline src/com/lazydroid/autoupdateapk/AutoUpdateApk.java]: yes but I prefer keeping things explicit, given this is a workaround until Google fixes the issue!

**Comment by lazydroid** [comment]
Thanks, Nicola!

**Comment by jofrantoba** [comment]
this lib is missing com.nbeghin.smartchandiser.BuildConfig

**Comment by nicolabeghin** [comment]
Just change it with your app's BuildConfig

**Pull request merged**